companion.py

```diff
@@ -545,176 +707,344 @@ def auth_callback(self) -> None:
+                    logger.warning(f"{last_starport_id!r} != {int(shipyard_data['id'])!r}")
+                    raise ServerLagging()
+
+                else:
+                    shipyard_data['name'] = last_starport_name
+                    station_data['lastStarport'].update(shipyard_data)
+            # WORKAROUND END
+
+            return station_data
+
+        while True:
+            query = self.capi_request_queue.get()
+            logger.trace_if('capi.worker', 'De-queued request')
+            if not isinstance(query, EDMCCAPIRequest):
+                logger.error("Item from queue wasn't an EDMCCAPIRequest")
+                break
+
+            if query.endpoint == query.REQUEST_WORKER_SHUTDOWN:
+                logger.info(f'endpoint {query.REQUEST_WORKER_SHUTDOWN}, exiting...')
+                break
 
-    def station(self) -> CAPIData:  # noqa: CCR001
+            logger.trace_if('capi.worker', f'Processing query: {query.endpoint}')
+            capi_data: CAPIData
+            if query.endpoint == self._CAPI_PATH_STATION:
+                try:
+                    capi_data = capi_station_queries()
+
+                except Exception as e:
+                    self.capi_response_queue.put(
+                        EDMCCAPIFailedRequest(
+                            message=str(e.args),
+                            exception=e,
+                            query_time=query.query_time,
+                            play_sound=query.play_sound,
+                            auto_update=query.auto_update
+                        )
+                    )
+
+                else:
+                    self.capi_response_queue.put(
+                        EDMCCAPIResponse(
+                            capi_data=capi_data,
+                            query_time=query.query_time,
+                            play_sound=query.play_sound,
+                            auto_update=query.auto_update
+                        )
+                    )
+
+            else:
+                try:
+                    capi_data = capi_single_query(self.FRONTIER_CAPI_PATH_PROFILE)
+
+                except Exception as e:
+                    self.capi_response_queue.put(
+                        EDMCCAPIFailedRequest(
+                            message=e.args,
+                            exception=e,
+                            query_time=query.query_time,
+                            play_sound=query.play_sound,
+                            auto_update=query.auto_update
+                        )
+                    )
+
+                else:
+                    self.capi_response_queue.put(
+                        EDMCCAPIResponse(
+                            capi_data=capi_data,
+                            query_time=query.query_time,
+                            play_sound=query.play_sound,
+                            auto_update=query.auto_update
+                        )
+                    )
```

Comment:
Im sure we can maybe mess this such that theres a single try/except/else ? possibly with a wrapper function?

Reply:
Possibly, and I am now wondering why I did `str(e.args)` in one place but not the other.

I think the real fix is to move the try/except clauses in an indent.  I'll play with that.